server/server.py

```diff
@@ -632,9 +637,18 @@ def handle_client_message(self, addr, message, room=None):
                         remaining_cooldown = 0
                         
                         if room.game.trains[nickname].boost_cooldown_active:
-                            current_time = time.time()
-                            elapsed_time = current_time - room.game.trains[nickname].start_cooldown_time
-                            remaining_cooldown = max(0, BOOST_COOLDOWN_DURATION - elapsed_time)
+                            # Use tick-based cooldown calculation
+                            current_tick = room.game.trains[nickname].move_timer
+                            ticks_elapsed = current_tick - room.game.trains[nickname].start_cooldown_tick
+                            
+                            # Convert duration to ticks using the same approach as in train.py
+                            standard_tickrate = 60.0  # Reference tickrate
+                            tickrate_ratio = standard_tickrate / room.game.trains[nickname].tick_rate
+                            required_ticks = int(BOOST_COOLDOWN_DURATION * room.game.trains[nickname].tick_rate * tickrate_ratio)
+                            
+                            remaining_ticks = max(0, required_ticks - ticks_elapsed)
+                            # Convert remaining ticks to seconds for user-friendly message
+                            remaining_cooldown = remaining_ticks / room.game.trains[nickname].tick_rate
```

Comment:
IMHO, the server should only deal with ticks. The client can convert to seconds for nicer messages.